File: valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int n=s.size();
        string str;
        if(s.empty()){
            return true;
        }
        for(int i=0;i<n;i++){
            if(isalnum(s[i])){
                str+=tolower(s[i]);
            }
        }
        int a=0, b=str.size()-1;
        while(a<b){
            if(str[a]!=str[b]){
                return false;
            }
            a++;b--;
        }
        return true;
    }
};

File: valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if(s.trim().isEmpty()){
            return true;
        }
        StringBuilder sb = new StringBuilder();
        for(char ch : s.toCharArray()){
            if(Character.isLetterOrDigit(ch)){
                sb.append(Character.toLowerCase(ch));
            }
        }
        return check(sb.toString());
    }
    public boolean check(String str){
        int i = 0, j = str.length() - 1;
        while( i <= j){
            if(str.charAt(i++) != str.charAt(j--)){
                return false;
            }
        }
        return true;
    }
}